### PR TITLE
Warm reboot: FDB  sync up

### PR DESCRIPTION
**Pull request by jipanyang**
Signed-off-by: Jipan Yang <jipan.yang@alibaba-inc.com>

**What I did**
During orchagent warm start window,  FDB may be changed. void FdbOrch::syncUpFdb() is to be called after the whole orchagent state has been restored.

**Why I did it**

**How I verified it**

VS test case to be provided once #554 is settled.

**Details if related**
Has dependency on #554

### CODE REVIEW

File: orchagent/fdborch.cpp [hunk 1]
```diff
@@ -36,6 +39,59 @@ FdbOrch::FdbOrch(DBConnector *db, string tableName, PortsOrch *port) :
     Orch::addExecutor("FDB_NOTIFICATIONS", fdbNotifier);
 }
 
+
+void FdbOrch::syncUpFdb()
```
Comment (zhenggen-xu) [inline orchagent/fdborch.cpp]: Where this syncUpFdb() is (going to be) called?

File: orchagent/fdborch.cpp [hunk 2]
```diff
@@ -36,6 +39,59 @@ FdbOrch::FdbOrch(DBConnector *db, string tableName, PortsOrch *port) :
...
+    for (const auto &key: keys)
+    {
+        std::vector<FieldValueTuple> values;
+
+        if (!table.get(key, values))
+        {
+            continue;
+        }
+
+        string s = tableName + ":" + key
```
Comment (zhenggen-xu) [inline orchagent/fdborch.cpp]: It would be better to not hardcode the separator.

File: orchagent/fdborch.cpp [hunk 3]
```diff
@@ -36,6 +36,56 @@ FdbOrch::FdbOrch(DBConnector *db, string tableName, PortsOrch *port) :
...
+    string tableName = "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY";
+    DBConnector *db = new DBConnector(ASIC_DB, DBConnector::DEFAULT_UNIXSOCKET, 0);
+    swss::Table table(db, tableName);
+    vector<string> keys;
+
+    table.getKeys(keys);
+    for (const auto &key: keys)
+    {
+        sai_object_id_t bridge_port_id = SAI_NULL_OBJECT_ID;
+        sai_fdb_entry_type_t entryType = SAI_FDB_ENTRY_TYPE_STATIC;
```
Comment (prsunny) [inline orchagent/fdborch.cpp]: IMO, the default can by DYNAMIC and check for STATIC in _if_ statement down considering most of the FDBs learnt are dynamic

File: orchagent/fdborch.cpp [hunk 4]
```diff
@@ -36,6 +36,56 @@ FdbOrch::FdbOrch(DBConnector *db, string tableName, PortsOrch *port) :
...
+        std::string value;
+
+        table.hget(key, "SAI_FDB_ENTRY_ATTR_TYPE", value);
+        if (value ==  "SAI_FDB_ENTRY_TYPE_DYNAMIC")
```
Comment (prsunny) [inline orchagent/fdborch.cpp]: Extra space after ==

**Comment by jipanyang** [comment]
@prsunny thanks for the review!   FDB sync up will require more sophisticated handling.  This version of change doesn't take FDB restore into account and assumes that no previous dynamic FDB   exists.   Besides new FDB entries,  there might be stale entries as well. 

Will prepare a new version to handle the case of stale FDB entries.

**Comment by qiluo-msft** [comment]
Closed, in favor of https://github.com/Azure/sonic-swss/pull/615
Thanks for the inspiration.

**Pull request closed without merging**